x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/entsearch/engine/Engine.java

```diff
@@ -152,6 +152,7 @@ public static Engine fromXContent(String resourceName, XContentParser parser) th
     @Override
     public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
         builder.startObject();
+        builder.field(NAME_FIELD.getPreferredName(), name);
```

Comment:
That won't work if we don't add this field in the rest parser too (see the failing tests in the PR). Although I am not sure we want to do that since we'd need to ensure that the name matches the id that is provided in the request. Can we remove it for now and discuss if this is really needed in a follow up?

Reply:
We could remove that, but then the API would need to be modified, as there would be no ID returned for each of the Engines returned (and thus we wouldn't be able to identify them).

I'll check adding them to the parser, happy to hear alternatives / better response formats, even though it might impact the frontend work.

Reply:
I think it would be more complete to add it to the response.

Reply:
Ok so let's add it as an optional param in the parser and we can validate that it matches the expected name when we build the Engine object? We should also do the same for the alias name since it's accepted by the parser currently but we don't check if the alias name is correct.

Reply:
Added in the parser, please check the solution 👍